File: nohealthcheck/autonomi-anttp-no-healthcheck/autonomi-anttp-fixed/backend/app/services/chunks.py
"""Chunk service for interacting with AntTP chunks API."""
import httpx
import base64
from typing import Optional, Any
from app.config import get_settings


class ChunkService:
    """Service for managing chunks on Autonomi Network."""
    
    def __init__(self, anttp_client: Optional[httpx.AsyncClient] = None):
        """Initialize chunk service.
        
        Args:
            anttp_client: Optional HTTP client for testing
        """
        self.settings = get_settings()
        self.client = anttp_client or httpx.AsyncClient(
            base_url=self.settings.anttp_base_url,
            timeout=self.settings.anttp_timeout
        )
    
    def _validate_base64(self, content: str) -> None:
        """Validate that content is valid base64.
        
        Args:
            content: Base64-encoded string
            
        Raises:
            ValueError: If content is not valid base64
        """
        try:
            base64.b64decode(content, validate=True)
        except Exception:
            raise ValueError("Invalid base64 content")
    
    async def create_chunk(
        self, 
        content: str, 
        store_type: str = "network"
    ) -> dict[str, Any]:
        """Create a new chunk with the given content.
        
        Args:
            content: Base64-encoded content
            store_type: Storage type (memory, disk, or network)
            
        Returns:
            Dict with address and store_type
            
        Raises:
            ValueError: If content is invalid
            Exception: If creation fails
        """
        # Validate input
        self._validate_base64(content)
        
        # Prepare request
        headers = {}
        if store_type in ("memory", "disk"):
            headers["x-cache-only"] = "true"
        
        payload = {
            "content": content
        }
        
        # Make request
        response = await self.client.post(
            "/api/v1/chunks",
            json=payload,
            headers=headers
        )
        
        if not response.is_success:
            raise Exception(f"Failed to create chunk: {response.status_code}")
        
        data = response.json()
        return {
            "address": data["address"],
            "store_type": store_type
        }
    
    async def get_chunk(self, address: str) -> dict[str, Any]:
        """Retrieve a chunk by its address.
        
        Args:
            address: Chunk address (XOR)
            
        Returns:
            Dict with address, content, and store_type
            
        Raises:
            Exception: If chunk not found or retrieval fails
        """
        response = await self.client.get(f"/api/v1/chunks/{address}")
        
        if response.status_code == 404:
            raise Exception(f"Chunk not found: 404")
        
        if not response.is_success:
            raise Exception(f"Failed to get chunk: {response.status_code}")
        
        data = response.json()
        return {
            "address": data["address"],
            "content": data.get("content"),
            "store_type": data.get("store_type", "unknown")
        }
    
    async def close(self):
        """Close the HTTP client."""
        await self.client.aclose()

File: nohealthcheck/autonomi-anttp-no-healthcheck/autonomi-anttp-fixed/frontend/src/lib/api/chunks.ts
/**
 * API client for chunks endpoints
 */
import axios, { type AxiosInstance } from 'axios';

export interface ChunkResponse {
  address: string;
  content?: string;
  store_type: string;
}

export class ChunksAPI {
  private client: AxiosInstance;

  constructor(baseURL: string) {
    this.client = axios.create({
      baseURL,
      timeout: 30000,
      headers: {
        'Content-Type': 'application/json'
      }
    });
  }

  async createChunk(content: string, storeType: string = 'network'): Promise<ChunkResponse> {
    const response = await this.client.post('/chunks', {
      content,
      store_type: storeType
    });
    return response.data;
  }

  async getChunk(address: string): Promise<ChunkResponse> {
    const response = await this.client.get(`/chunks/${address}`);
    return response.data;
  }
}

/**
 * Main API client aggregating all endpoints
 */
export class APIClient {
  public chunks: ChunksAPI;
  
  constructor(baseURL: string = 'http://localhost:8000/api/v1') {
    this.chunks = new ChunksAPI(baseURL);
  }
}

// Export singleton instance
export const apiClient = new APIClient(
  import.meta.env.VITE_API_BASE_URL || 'http://localhost:8000/api/v1'
);
